pkg/downloader/manager.go

```diff
@@ -78,11 +78,6 @@ func (m *Manager) Build() error {
 		return m.Update()
 	}
 
-	req := c.Metadata.Dependencies
-	if sum, err := resolver.HashReq(req); err != nil || sum != lock.Digest {
-		return errors.New("Chart.lock is out of sync with Chart.yaml")
-	}
```

Comment:
Consider the case where a dependency has a range set on it of `^1.2.3` in the Chart.yaml file. Now, someone changes that range to `^1.3.4`. But, the lock file has not been updated since this change happened. The current code should catch that. Removing this section would stop the end user from knowing that something is wrong.

We need to fix the broken cases but I would want to warn the user if something is wrong.